mlir/include/mlir/Analysis/Presburger/IntegerRelation.h

```diff
@@ -711,6 +711,13 @@ class IntegerRelation {
   /// return `this \ set`.
   PresburgerRelation subtract(const PresburgerRelation &set) const;
 
+  // Remove equalities which have only zero coefficients.
+  void removeTrivialEqualities();
```

Comment:
Isn't this already implemented somewhere in IntegerRelation?

Reply:
@Superty suggested incorporating this in `removeTrivialRedundancy()`, do you mean that?